pkg/apis/core/types_shoot.go

```diff
@@ -523,6 +523,34 @@ type KubeAPIServerConfig struct {
 	EnableAnonymousAuthentication *bool
 	// EventTTL controls the amount of time to retain events.
 	EventTTL *metav1.Duration
+	// AccessControl provides authorization mechanisms for Shoot-API-Server.
+	AccessControl *AccessControl
+}
+
+// AccessControl provides authorization mechanisms for Shoot-API-Server.
+// Note: The schema (incl. child structs) and documentation resembles istio's AuthorizationPolicy, but is heavily simplified.
+type AccessControl struct {
+	// The action to take on the source of request.
+	Action AuthorizationAction
+	// Origin of request to run defined authorization action against.
+	Source Source
+}
+
+// AuthorizationAction is the operation (e.g. DENY) to apply on requests.
+type AuthorizationAction string
+
+const (
+	// Allows the request from a source.
+	AuthorizationAction_ALLOW AuthorizationAction = "ALLOW"
```

Comment:
Why CamelCase mixed with Snake_Case? 
```suggestion
	AuthorizationActionAllow AuthorizationAction = "ALLOW"
```

also, should we rename this into `AccessAction*` in the light of above?